### PR TITLE
Modular computers now use list of components

### PR DESCRIPTION
**Pull request by Razharas**
Modular computers now use list of components instead of hardcoded vars
Hardware now have device_type var that will be the key of assoc list in the computer by which to access the hardware part
Device types are textual but use defines
Adding hardware adds the verbs to eject stuff from computers instead of it being there all the time

### CODE REVIEW

File: code/modules/modular_computers/computers/item/computer_ui.dm [hunk 1]
```diff
@@ -112,7 +115,9 @@
 				update_icon()
 				return
 
-			if(idle_threads.len >= processor_unit.max_idle_programs+1)
+			var/obj/item/weapon/computer_hardware/processor_unit/PU = all_components[MC_CPU]
+
+			if(idle_threads.len >= PU.max_idle_programs+1)
```
Comment (RemieRichards) [inline code/modules/modular_computers/computers/item/computer_ui.dm]: ```if(idle_threatds.len > PU.max_idle_programs)```
much simpler, no +1 fuckery.

Comment (Razharas) [inline code/modules/modular_computers/computers/item/computer_ui.dm]: Why wasnt this fixed durning porting is the question

Comment (Core0verload) [inline code/modules/modular_computers/computers/item/computer_ui.dm]: Nothing was fixed durning porting, it was ported as is.

**Review by phil235 — APPROVED** [review]

File: code/modules/modular_computers/computers/item/computer_ui.dm (hunk 1 above)
Comment (Shadowlight213) [inline code/modules/modular_computers/computers/item/computer_ui.dm]: Yup. I Had 0 clue how most of it worked, so I mainly just focused on the interface and nanoui -> tgui stuff.

**Pull request merged**